test/e2e/conftest.py

```diff
@@ -334,8 +342,19 @@ def terminology(workflow_api, dataplane_api, stack_resources, testing_env_variab
     assert create_terminology_request.status_code == 200
 
     # wait for terminology to complete
-    # FIXME - should us a polling loop here
-    time.sleep(60)
+
+    processing = True
+
+    while create_terminology_request.status_code == 200 and processing:
+        body = {"terminology_name": "uitestterminology"}
+        get_terminology_response = workflow_api.get_terminology_request(body)
+
+        if get_terminology_response.status_code == 200:
```

Comment:
This loop looks like it could potentially run infinitely.

* Are there HTTP Response codes that we should not retry?
* Should we have a built in timeout or attempt limit such that, once exhausted, we give up trying again and fail the test?
* Does a response code `200` always mean it is ready or is there some other state we need to verify in the response?

Reply:
the [documentation](https://docs.aws.amazon.com/translate/latest/APIReference/API_GetTerminology.html#API_GetTerminology_ResponseElements) only mentions 200 as successful